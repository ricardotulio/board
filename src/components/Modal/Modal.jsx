import React from 'react'
import styles from './styles.module.css'

const Modal = ({ children, active = false }) => (
  <div
    className={[styles.modal, (active ? styles.active : [])].join(' ')}
  >
    <div className={styles.content}>
      <span className={styles.close}>&times;</span>
      <div>{ children }</div>
    </div>
  </div>
)

export default Modal
