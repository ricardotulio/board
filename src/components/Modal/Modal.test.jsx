import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import Modal from './Modal'

it('can render a modal', () => {
  const { container } = render(<Modal />)
  expect(container.firstChild).toMatchSnapshot()
})

it('should start invisible', () => {
  const { container } = render(<Modal />)
  expect(container.firstChild).toHaveClass('modal')
  expect(container.firstChild.classList.contains('active')).toBe(false)
})

it('should be visible if is activated', () => {
  const { container } = render(<Modal active={ true } />)
  expect(container.firstChild).toHaveClass('modal')
  expect(container.firstChild).toHaveClass('active')
})
