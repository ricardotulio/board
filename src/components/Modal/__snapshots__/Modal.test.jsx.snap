// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can render a modal 1`] = `
<div
  class="modal "
>
  <div
    class="content"
  >
    <span
      class="close"
    >
      Ã—
    </span>
    <div />
  </div>
</div>
`;
